src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.php

```diff
@@ -209,4 +210,11 @@
                     ->args([service('request_stack'), param('kernel.debug')]),
             ])
     ;
+
+    if (class_exists(\BackedEnum::class) || interface_exists(\BackedEnum::class) || trait_exists('\BackedEnum')) {
```

Comment:
can be simplified. This must not be a trait (which won't work for instanceof) and I'm quick sure that this is always an interface

Reply:
I agree for the trait. About the class, what if someone creates a class polyfill and not an interface, as @OskarStark said?

Reply:
well, if you actually want to create a polyfill, `BackedEnum` itself should be an interface. Otherwise, that's not actually a polyfill.

Reply:
My bad, you're right. Thanks for your explanation, fixing this!